sdk/auth/azure_cli_authorizer.go

```diff
@@ -199,11 +218,34 @@ func newAzureCliConfig(api environments.Api, tenantId string, auxiliaryTenantIds
 		subscriptionId = *defaultSubscriptionId
 	}
 
+	// validate subscriptionIdHint, if applicable (currently only for Resource Manager)
+	if environments.ApiIsKnown(api, "AzureResourceManager") {
+		if subscriptionIdHint != "" {
+			if availableSubscriptionIds, err := azurecli.ListAvailableSubscriptionIDs(); err != nil {
+				return nil, err
+			} else if availableSubscriptionIds == nil {
+				return nil, fmt.Errorf("no available subscription IDs returned by Azure CLI")
+			} else {
+				found := false
+				for _, subId := range *availableSubscriptionIds {
+					if strings.EqualFold(subId, subscriptionIdHint) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return nil, fmt.Errorf("the provided subscription ID %q is not known by Azure CLI", subscriptionIdHint)
+				}
+			}
+		}
+	}
+
 	return &azureCliConfig{
 		Api:                   api,
 		TenantID:              tenantId,
 		AuxiliaryTenantIDs:    auxiliaryTenantIds,
 		DefaultSubscriptionID: subscriptionId,
+		SubscriptionIDHint:    strings.ToLower(subscriptionIdHint),
```

Comment:
lower-casing the Subscription ID _feels_ like it'd be potentially problematic?

Reply:
I think we should move to normalize this wherever possible, although az-cli and the API accept any casing here. In this case there's no knock-on effect since this field isn't consumed/referenced anywhere, except when acquiring an access token.

Reply:
👍 as discussed offline, this instance is fine - we'll want to take this into account elsewhere too